src/js/eventInTimeAware.js

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.txt
+*/
+
+/* global fluid, floe */
+
+(function ($, fluid) {
+
+    "use strict";
+
+    // A grade that deals in time,
```

Comment:
Incredibly inadequate comment : P
This should explain - i) what fields are created in the model, their purposes and relationships, ii) what expectations there are on the interactions of these with the rest of the model - I notice that there is a modelListener attached to "" which is pretty aggressive and will strongly inhibit the reuse of this component. Note that with the implementation of https://issues.fluidproject.org/browse/FLUID-5695 we have support for IoC-resolved paths and path segments, which we should use here to make this component more reusable